src/main/java/io/github/opencubicchunks/cubicchunks/core/server/chunkio/RegionCubeStorage.java

```diff
@@ -0,0 +1,198 @@
+                                            .build(),
+                                    (dir, key) -> Files.exists(dir.resolve(key.getRegionKey().getName()))
+                            )
+                    ),
+                    new SharedCachedRegionProvider<>(
+                            new SimpleRegionProvider<>(new EntryLocation2D.Provider(), part2d,
+                                    (keyProvider, regionKey) -> new ExtRegion<>(part2d, Collections.emptyList(), keyProvider, regionKey),
+                                    (dir, key) -> Files.exists(dir.resolve(key.getRegionKey().getName() + ".ext"))
+                            )
+                    ));
+            SaveSection3D section3d = new SaveSection3D(
+                    new SharedCachedRegionProvider<>(
+                            new SimpleRegionProvider<>(new EntryLocation3D.Provider(), part3d, (keyProv, r) ->
+                                    ShadowPagingRegion.<EntryLocation3D>builder()
+                                            .setDirectory(part3d)
+                                            .setRegionKey(r)
+                                            .setKeyProvider(keyProv)
+                                            .setSectorSize(512)
+                                            .build(),
+                                    (dir, key) -> Files.exists(dir.resolve(key.getRegionKey().getName()))
+                            )
+                    ),
+                    new SharedCachedRegionProvider<>(
+                            new SimpleRegionProvider<>(new EntryLocation3D.Provider(), part3d,
+                                    (keyProvider, regionKey) -> new ExtRegion<>(part3d, Collections.emptyList(), keyProvider, regionKey),
+                                    (dir, key) -> Files.exists(dir.resolve(key.getRegionKey().getName() + ".ext"))
+                            )
+                    ));
+
+            return new SaveCubeColumns(section2d, section3d);
+        } else {
+            return SaveCubeColumns.create(path);
+        }
+    }
+
+    private final Path path;
+    private SaveCubeColumns save;
+
+    public RegionCubeStorage(Path path) throws IOException {
+        this.path = Objects.requireNonNull(path, "path");
+        this.save = saveForPath(path);
+    }
+
+    @Override
+    public boolean columnExists(ChunkPos pos) throws IOException {
+        return this.save.getSaveSection2D().hasEntry(new EntryLocation2D(pos.x, pos.z));
+    }
+
+    @Override
+    public boolean cubeExists(CubePos pos) throws IOException {
+        return this.save.getSaveSection3D().hasEntry(new EntryLocation3D(pos.getX(), pos.getY(), pos.getZ()));
+    }
+
+    @Override
+    public NBTTagCompound readColumn(ChunkPos pos) throws IOException {
+        //TODO: i'm pretty sure this should be false
+        Optional<ByteBuffer> data = this.save.load(new EntryLocation2D(pos.x, pos.z), true);
```

Comment:
Iirc the reason I used `true` here is that when using `false`, RegionLib won't cache the region file (as there is nothing to cache) and do a Files.exists() check on every load request until a region is written, which is horribly slow on windows. You can put this explanation in a comment.